packages/cli/lib/services/deploy.service.ts

```diff
@@ -364,18 +365,27 @@ class DeployService {
         const onEventScriptsByProvider: OnEventScriptsByProvider[] = [];
 
         for (const integration of parsed.integrations) {
-            const { providerConfigKey, postConnectionScripts } = integration;
-
-            if (postConnectionScripts && postConnectionScripts.length > 0) {
-                const scripts: OnEventScriptsByProvider['scripts'] = [];
-                for (const postConnectionScript of postConnectionScripts) {
-                    const files = loadScriptFiles({ scriptName: postConnectionScript, providerConfigKey, fullPath, type: 'post-connection-scripts' });
-                    if (!files) {
-                        return null;
+            const { providerConfigKey, onEventScripts, postConnectionScripts } = integration;
+
+            const scripts: OnEventScriptsByProvider['scripts'] = [];
+            for (const event in onEventScripts || {}) {
+                const e = event as OnEventType;
```

Comment:
Why it needs type correction? Also `e` could be better :D

Reply:
https://github.com/microsoft/TypeScript/issues/32811

Reply:
I will rename